tests/expr_and_series/fill_null_test.py

```diff
@@ -24,17 +30,268 @@ def test_fill_null(constructor: Constructor) -> None:
     assert_equal_data(result, expected)
 
 
+def test_fill_null_exceptions(constructor: Constructor) -> None:
+    data = {
+        "a": [0.0, None, 2, 3, 4],
+    }
+    df = nw.from_native(constructor(data))
+
+    with pytest.raises(ValueError, match="cannot specify both `value` and `strategy`"):
+        df.with_columns(nw.col("a").fill_null(value=99, strategy="forward"))
+
+    with pytest.raises(
+        ValueError, match="must specify either a fill `value` or `strategy`"
+    ):
+        df.with_columns(nw.col("a").fill_null())
+
+    with pytest.raises(ValueError, match="strategy not supported:"):
+        df.with_columns(nw.col("a").fill_null(strategy="invalid"))  # type: ignore  # noqa: PGH003
+
+
+def test_fill_null_strategies_with_limit_as_none(constructor: Constructor) -> None:
+    data_limits = {
+        "a": [1, None, None, None, 5, 6, None, None, None, 10],
+        "b": ["a", None, None, None, "b", "c", None, None, None, "d"],
+    }
+    df = nw.from_native(constructor(data_limits))
+
+    expected_forward = {
+        "a": [1, 1, 1, 1, 5, 6, 6, 6, 6, 10],
+        "b": ["a", "a", "a", "a", "b", "c", "c", "c", "c", "d"],
+    }
+    if (
+        "pandas_pyarrow_constructor" in str(constructor)
+        or "modin" in str(constructor)
+        or "dask" in str(constructor)
+    ):
+        with warnings.catch_warnings():
+            warnings.simplefilter(action="ignore", category=pd.errors.PerformanceWarning)
+            warnings.simplefilter(action="ignore", category=FutureWarning)
```

Comment:
this is too broad, and will potentially catch too much - could you just use `warnings.filter` instead, and match on the warning message?

Reply:
Done!